src/backend/renderer/mod.rs

```diff
@@ -602,6 +602,217 @@ pub fn buffer_type(buffer: &wl_buffer::WlBuffer) -> Option<BufferType> {
+                | wl_shm::Format::Yuv410
+                | wl_shm::Format::Yvu410
+                | wl_shm::Format::Yuv411
+                | wl_shm::Format::Yvu411
+                | wl_shm::Format::Yuv420
+                | wl_shm::Format::Yvu420
+                | wl_shm::Format::Yuv422
+                | wl_shm::Format::Yvu422
+                | wl_shm::Format::Yuv444
+                | wl_shm::Format::Yvu444
+                | wl_shm::Format::R8
+                | wl_shm::Format::R16
+                | wl_shm::Format::Rg88
+                | wl_shm::Format::Gr88
+                | wl_shm::Format::Rg1616
+                | wl_shm::Format::Gr1616
+                | wl_shm::Format::Xrgb16161616f
+                | wl_shm::Format::Xbgr16161616f
+                | wl_shm::Format::Xyuv8888
+                | wl_shm::Format::Vuy888
+                | wl_shm::Format::Vuy101010
+                | wl_shm::Format::Y210
+                | wl_shm::Format::Y212
+                | wl_shm::Format::Y216
+                | wl_shm::Format::Y410
+                | wl_shm::Format::Y412
+                | wl_shm::Format::Y416
+                | wl_shm::Format::Xvyu2101010
+                | wl_shm::Format::Xvyu1216161616
+                | wl_shm::Format::Xvyu16161616
+                | wl_shm::Format::Y0l0
+                | wl_shm::Format::X0l0
+                | wl_shm::Format::Y0l2
+                | wl_shm::Format::X0l2
+                | wl_shm::Format::Yuv4208bit
+                | wl_shm::Format::Yuv42010bit
+                | wl_shm::Format::Xrgb8888A8
+                | wl_shm::Format::Xbgr8888A8
+                | wl_shm::Format::Rgbx8888A8
+                | wl_shm::Format::Bgrx8888A8
+                | wl_shm::Format::Rgb888A8
+                | wl_shm::Format::Bgr888A8
+                | wl_shm::Format::Rgb565A8
+                | wl_shm::Format::Bgr565A8
+                | wl_shm::Format::Nv24
+                | wl_shm::Format::Nv42
+                | wl_shm::Format::P210
+                | wl_shm::Format::P010
+                | wl_shm::Format::P012
+                | wl_shm::Format::P016
+                | wl_shm::Format::Nv15
+                | wl_shm::Format::Q410
+                | wl_shm::Format::Q401
+                | wl_shm::Format::Xrgb16161616
+                | wl_shm::Format::Xbgr16161616
+        ));
+    }
+
+    // Not managed, check if this is an EGLBuffer
+    #[cfg(all(feature = "backend_egl", feature = "use_system_lib"))]
+    if let Some(format) = BUFFER_READER
+        .lock()
+        .unwrap()
+        .as_ref()
+        .and_then(|x| x.upgrade())
+        .and_then(|x| x.egl_buffer_contents(buffer).ok())
+        .map(|x| x.format.clone())
+    {
+        return Some(!matches!(
+            format,
+            super::egl::Format::RGB
+                | super::egl::Format::Y_UV
+                | super::egl::Format::Y_U_V
+                | super::egl::Format::Y_XUXV
+        ));
+    }
```

Comment:
And I guess this is short enough, that it could stay. But potentially this could also be a `has_alpha` function of the `EGLBufferReader`.

Reply:
Added `egl_buffer_has_alpha` to the reader